packages/serverless-orchestration/src/ServerlessHub.js

```diff
@@ -397,15 +398,23 @@ async function _getLastQueriedBlockNumber(configIdentifier, chainId, logger) {
   );
 }
 
+function _getWeb3AndUrlForBot(botConfig) {
+  const retryConfig = botConfig?.environmentVariables?.NODE_RETRY_CONFIG;
+  if (retryConfig) {
+    return [new Web3(createBasicProvider(retryConfig)), retryConfig[0].url];
```

Comment:
hmm. you are taking the 0th element from this array. does this not mean that you are only creating the web3 instance from the first provider and therefore are not creating a re-try provider, using all of the provided node URLs?

Reply:
No, previously we were using the url as the key to cache information. To keep that structure, I just use the first url as the "key". The web3 being used to make the calls is the first element of the return.

Does that make sense?

Reply:
(Happy to go with a different function or key structure if you have a different preference!!)

Reply:
O I see, ye I missed that. thanks for explaining, makes sence!